gcc/config/riscv/riscv.h

```diff
@@ -513,15 +513,15 @@ enum reg_class
    `crtl->outgoing_args_size'.  */
 #define OUTGOING_REG_PARM_STACK_SPACE(FNTYPE) 1
 
-#define STACK_BOUNDARY 128
+#define STACK_BOUNDARY (TARGET_RVE ? 32 : 128)
 
 /* Symbolic macros for the registers used to return integer and floating
    point values.  */
 
 #define GP_RETURN GP_ARG_FIRST
 #define FP_RETURN (UNITS_PER_FP_ARG == 0 ? GP_RETURN : FP_ARG_FIRST)
 
-#define MAX_ARGS_IN_REGISTERS 8
+#define MAX_ARGS_IN_REGISTERS (TARGET_RVE ? 6 : 8)
```

Comment:
Do you mind submitting a patch to https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md that describes the E calling convention?